fix: OPTIC-1748: [Timebox] UI polish

yyassi-heartex opened the pull request:
<!-- Follow Merge -->
This PR was automatically generated via Follow Merge. [Workflow Run](https://github.com/HumanSignal/label-studio/actions/runs/13763179195)
Please ensure that all linked upstream Pull Requests are merged before proceeding with this one.
- 
<!-- Follow Merge -->



various UI improvements and polish including:

- tooltip and user pic component unification
  - this is to address inconsistencies in usage and visuals for these components
- text change on the Order by dropdown
- List/Grid view toggle is now an icon
- Project card, status icon tooltips
- fixed an issue affecting datamanger checkboxes when all columns are selected to display, causing them to be duplicated

### PR fulfills these requirements
- [x] Commit message(s) and PR title follows the format `[fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made` ex. `fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors`
- [ ] Tests for the changes have been added/updated (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [x] Self-reviewed and ran all changes on a local instance (for bug fixes/features)



#### Change has impacts in these area(s)
_(check all that apply)_
- [ ] Product design
- [ ] Backend (Database)
- [ ] Backend (API)
- [x] Frontend

This pull request includes multiple changes aimed at refactoring the codebase, specifically focusing on updating import paths and removing deprecated components. The most important changes are grouped by theme below:

### Refactoring Import Paths:
* Updated import paths for `Tooltip` to use `@humansignal/ui` in various components, such as `CopyableTooltip`, `DescriptionList`, `HeidiTip`, `MachineLearningList`, `PredictionsList`, and `StorageSummary`. [[1]](diffhunk://#diff-152fdc46a745919660f4031569955baec2bbf8d873e5cdc8097aa85831601f5dL3-R3) [[2]](diffhunk://#diff-6016147ee27fdf712c39b880eec5be50b0822fb22381a878ed7689501863cd8cL4-R4) [[3]](diffhunk://#diff-6c60f330dfea598fa1ef72af9e384b37745f2878caac5fabc4369a9c6a0c67b4L8-R8) [[4]](diffhunk://#diff-14a94a9820688f85db09863f58c40380606751614befd1378117aac863c73f59L9-R9) [[5]](diffhunk://#diff-e20e76212c5d53c2e7ec74919fe73d4ee27109d08d32039da77ae2746ebe3464L6-R6) [[6]](diffhunk://#diff-466e193f97b7964093f7265d6b60555d477712bac726e7ce754e7814f38868b2L4-R4)

* Updated import paths for `alignElements` and `aroundTransition` to use `@humansignal/core/lib/utils/dom` and `@humansignal/core/lib/utils/transition` respectively in `DropdownComponent` and `ModalPopup`. [[1]](diffhunk://#diff-78f827eb6ecd9b2dcba399bca9efe8ac5e319b49a419bfe5ec45c14d4188eb7fL4-R5) [[2]](diffhunk://#diff-6f384ee14bef129371794a8d7942469d1554f96a68efc23ff89e32a5781d6683L5-R5)

* Updated import paths for `Userpic` to use `@humansignal/ui` in various components, such as `Menubar`, `PeopleList`, `SelectedUser`, and `ProjectsList`. [[1]](diffhunk://#diff-e7879b6adc8fea7d0c6bbe05245ab6dc5dc66f11347b27a2e3842ce36ade779cL24-R24) [[2]](diffhunk://#diff-66f63029b747881162c62d1fd9fc25e6d4f11769695896a74a42ef1910677410L3-R4) [[3]](diffhunk://#diff-d24fbe1921db75d7364927dc6b308975238f38261c741d044f9f0d97150af954L3-R4) [[4]](diffhunk://#diff-f57b59e4db0ae3a13e1b77902ad47a67e4683b7efd8971155448d6e39d49e9d8R5-R7)

### Removal of Deprecated Components:
* Removed the `Tooltip` component and its associated styles from the codebase, as it has been replaced by the version from `@humansignal/ui`. [[1]](diffhunk://#diff-e21001bd8e833d15350f97e3add7f5b9d18e8a096c0684f503a58520c8dee913L1-L114) [[2]](diffhunk://#diff-240bbaffdee76ab3b870631b605c2e943224df4bb2047d31b6f3971142eeb51fL1-L84)

* Removed the `Userpic` component and its associated styles from the codebase, as it has been replaced by the version from `@humansignal/ui`. [[1]](diffhunk://#diff-2881b55d7aa673f95e976ecde751054d6a86ab69fd7cb861454291d74276f785L1-L84) [[2]](diffhunk://#diff-59aa206efd092831b516099550f184eae851a5f64ec257ee35ca8bb27ffc9cddL1-L31)

* Removed the `alignElements` utility function from `web/apps/labelstudio/src/utils/dom.js` as it has been moved to `@humansignal/core/lib/utils/dom`.

### Codebase Simplification:
* Simplified the `getBoundingBox` and `getAbsolutePosition` functions in the renamed file `web/libs/core/src/lib/utils/dom.ts` to ensure consistent element positioning logic. [[1]](diffhunk://#diff-5bcd905e2119e59538abf86ccee7613a1c338b995755a2dd89dd950164742bb6R1-R40) [[2]](diffhunk://#diff-5bcd905e2119e59538abf86ccee7613a1c338b995755a2dd89dd950164742bb6L30-R68)

These changes collectively aim to streamline the codebase by consolidating utility functions and components into shared libraries, reducing redundancy, and improving maintainability.

yyassi-heartex committed:
fix: OPTIC-1748: [Timebox] UI polish

yyassi-heartex committed:
changing Delete back to Remove

yyassi-heartex committed:
standardizing utils/dom

yyassi-heartex committed:
fixing an issue that affects columns on datamanger table

yyassi-heartex committed:
lint cleanup

yyassi-heartex committed:
more cleanup

yyassi-heartex committed:
tweaking tooltip top-right positioning + view toggle switch to icons + comment item tooltip fixes

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1748/ui-improvements

yyassi-heartex committed:
lint cleanup

yyassi-heartex committed:
unifying Userpic

yyassi-heartex committed:
classname fix for Userpic

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1748/ui-improvements

yyassi-heartex committed:
adding test ids and updating how we check for toggle

yyassi-heartex committed:
lint cleanup

yyassi-heartex committed:
fixing column check

yyassi-heartex committed:
lint cleanup

yyassi-heartex committed:
fixed unit test + reduced padding on datamanger view toggle buttons

yyassi-heartex committed:
lint + test fixes

yyassi-heartex committed:
test fixes

yyassi-heartex committed:
addressing issues around datamanager columns checkboxes

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1748/ui-improvements

yyassi-heartex committed:
addressing failing checks

yyassi-heartex committed:
fixing userDisplayName

yyassi-heartex committed:
lint cleanup

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1748/ui-improvements

bmartel committed:
fixing AnnotationsCarousel tests

yyassi-heartex committed:
cleaning up a situation where its possible for user to be null when we try to spread its values

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1748/ui-improvements

nikitabelonogov commented:
/fm sync

nikitabelonogov commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13697081739)

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-optic-1748/ui-improvements'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13697081739

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13697081739

borisheartex commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13697520796)

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13697520796

yyassi-heartex committed:
Merge branch 'fb-optic-1748/ui-improvements' of github.com:heartexlabs/label-studio into fb-optic-1748/ui-improvements

yyassi-heartex committed:
fixing an issue that makes it impossible for a user to update the fields

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1748/ui-improvements

yyassi-heartex committed:
Revert "Merge remote-tracking branch 'origin/develop' into fb-optic-1748/ui-improvements"

This reverts commit 291f6e37083861000c5d3bbb1ff2e73e7119a2ce, reversing
changes made to 1c53cd9462bceb87ee0d9a499be0ec0b1410dce1.

yyassi-heartex committed:
Reapply "Merge remote-tracking branch 'origin/develop' into fb-optic-1748/ui-improvements"

This reverts commit 233340c6576e1af660ec6386eb45c2ac65e6d144.

yyassi-heartex committed:
fixing merge

yyassi-heartex committed:
cleaning up imports

AndrejOros commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13716495646)

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13716495646

yyassi-heartex commented on web/libs/core/src/pages/AccountSettings/sections/PersonalInfo.tsx:
```diff
@@ -75,6 +82,7 @@ export const PersonalInfo = () => {
       if (!user) return;
       const body = new FormData(e.currentTarget as HTMLFormElement);
       const json = Object.fromEntries(body.entries());
+      console.log("json", json);
```
```suggestion
```

yyassi-heartex replied:
```suggestion
```

yyassi-heartex commented on web/libs/ui/src/lib/Userpic/Userpic.tsx:
```diff
@@ -1,72 +1,84 @@
   faded?: boolean;
   showUsername?: boolean;
-  size?: any;
+  size?: number | false;
```
```suggestion
  size?: number;
```

yyassi-heartex commented on web/libs/ui/src/lib/Userpic/Userpic.tsx:
```diff
@@ -1,72 +1,84 @@
       faded = false,
       showUsername,
-      size,
+      size = false,
```
```suggestion
      size,
```

yyassi-heartex committed:
Update web/libs/core/src/pages/AccountSettings/sections/PersonalInfo.tsx

yyassi-heartex committed:
Update web/libs/ui/src/lib/Userpic/Userpic.tsx

yyassi-heartex committed:
Update web/libs/ui/src/lib/Userpic/Userpic.tsx

yyassi-heartex committed:
Merge branch 'develop' into fb-optic-1748/ui-improvements

yyassi-heartex committed:
cleaned out unused files + fixed tooltip breakage due to quick toggle + added intent delay

yyassi-heartex committed:
lint cleanup

AndrejOros commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13763179195)

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13763179195

yyassi-heartex commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13767500884)
> Successfully merged:  create mode 100644 web/apps/labelstudio/src/utils/query-client.ts

yyassi-heartex committed:
Merge branch 'develop' into 'fb-optic-1748/ui-improvements'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13767500884
